src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -654,17 +654,9 @@ public bool ApplyNullableTransforms(ImmutableArray<bool> transforms, INonNullTyp
                 result = result.WithTypeAndModifiers(newTypeSymbol, result.CustomModifiers);
             }
 
-            if (!result.IsValueType)
-            {
-                if (isAnnotated)
-                {
-                    result = result.AsNullableReferenceType();
-                }
-                else
-                {
-                    result = result.AsNotNullableReferenceType();
-                }
-            }
+            result = isAnnotated ?
+                result.AsNullableReferenceType() :
```

Comment:
Not related to this PR: we should probably rename these methods (`WithIsAnnotated`?) at some point.

Reply:
Added comment.

---
In reply to: [207445706](https://github.com/dotnet/roslyn/pull/29045#discussion_r207445706) [](ancestors = 207445706)